File: 38. Count and Say/CountAndSay.js
var countAndSay = function(n) {
    if (n === 1) {
        return '1';
    } else {
        num = countAndSay(n-1);
        temp = num[0];
        count = 0;
        result = '';
        for (let i = 0; i < num.length; i++) {
            if (temp === num[i]) {
                count++;
            } else {
                result += count + temp;
                count = 1;
                temp = num[i];
            }    
        }
        result += count + temp;
        return result;
    } 
};

File: 0038. Count and Say/CountAndSay.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if (n == 1):
            return '1'
        else:
            num = self.countAndSay(n - 1);
            s = ''
            prev = ''
            count = 0
            for i in num:
                if prev == i:
                    count += 1
                else:
                    if prev != '':
                        s += str(count) + prev
                    count = 1
                    prev = i
            s += str(count) + prev
            return s
